core/src/test/java/org/elasticsearch/search/child/ChildQuerySearchIT.java

```diff
@@ -1926,4 +1927,51 @@ public void testHasChildInnerQueryType() {
                 QueryBuilders.hasChildQuery("child-type", new IdsQueryBuilder().addIds("child-id"))).get();
         assertSearchHits(searchResponse, "parent-id");
     }
+
```

Comment:
@martijnvg can we do this with a unittest `HasChildQueryBuilderTests` should be able to verify this?